dashboards/cls_dashboard.py

```diff
@@ -0,0 +1,159 @@
+    sns.scatterplot(x=X_reduced[:, 0], y=X_reduced[:, 1], hue=labels)
+    st.pyplot(fig)
+
+def plot_confusion_matrix(cm: np.ndarray):
+    fig = plt.figure(figsize=(10, 10))
+    sns.heatmap(cm, annot=True)
+    st.pyplot(fig)
+
+
+def evaluate_classification(
+    model: RobertaForMaskedLM,
+    dataset_name: str,
+    quantizer: MidiQuantizer,
+    tokenizer: QuantizedMidiEncoder,
+    queries: list[str],
+    device: torch.device,
+):
+    dataloader = preprocess_dataset(
+        dataset_name,
+        quantizer=quantizer,
+        tokenizer=tokenizer,
+        queries=queries,
+        batch_size=1024,
+        num_workers=8,
+    )
+    query_to_id = {k: i for i, k in enumerate(queries)}
+
+    outputs = []
+    labels = []
+
+    # velocity time encoding
+    for batch in dataloader:
+        with torch.no_grad():
+            source = batch["source"]
+            input_token_ids = batch["input_token_ids"].to(device)
+
+            # shape [batch_size, seq_len, vocab_size]
+            out = model(input_ids=input_token_ids).logits
+            cls_embedding = out[:, 0, :]
+            outputs.append(cls_embedding)
+
+            for s in source:
+                for query in queries:
+                    if str.lower(query) in str.lower(s):
+                        labels.append(query)
+
+    # shape: [num_sequences, vocab_size]
+    X = torch.cat(outputs, dim=0).cpu().numpy()
+    y = np.array([query_to_id[q] for q in labels])
+
+    assert len(X) == len(y)
+
+    plot_embeddings(X, labels)
+
+    # reducing dimensions
+    pca = PCA(n_components=128)
+    X_reduced = pca.fit_transform(X)
+
+    X_train, X_test, y_train, y_test = train_test_split(X_reduced, y, test_size=0.2, random_state=42)
+
+    tree = DecisionTreeClassifier()
+    tree.fit(X_train, y_train)
+
+    acc = tree.score(X_test, y_test)
+    cm = confusion_matrix(tree.predict(X_test), y_test)
+    plot_confusion_matrix(cm)
+
+    return acc
+
+
+def main():
+    checkpoint = torch.load(
+        "checkpoints/masked-midi-modelling-2023-10-15-10-19-params-90.31M.ckpt"
+        # "checkpoints/pianomask_2023_10_08_19_10.pt"
+    )
+
+    cfg = checkpoint["config"]
+    device = cfg.train.device
+
+    quantizer = MidiQuantizer(
+        n_dstart_bins=cfg.quantization.dstart_bin,
+        n_duration_bins=cfg.quantization.duration_bin,
+        n_velocity_bins=cfg.quantization.velocity_bin,
+    )
+    tokenizer = QuantizedMidiEncoder(
+        dstart_bin=cfg.quantization.dstart_bin,
+        duration_bin=cfg.quantization.duration_bin,
+        velocity_bin=cfg.quantization.velocity_bin,
+    )
+
+    model_config = RobertaConfig(vocab_size=tokenizer.vocab_size)
+    model = RobertaForMaskedLM(model_config).to(device)
+
+    model.load_state_dict(checkpoint["model"])
+    # model.load_state_dict(checkpoint["model_state_dict"])
+    model.eval()
+
+    dataset_name = "JasiekKaczmarczyk/maestro-v1-sustain-masked"
+
+    queries_txt = st.text_input(label="Queries")
```

Comment:
```suggestion
    queries_txt = st.text_input(label="Queries", value="chopin, bach")
```

Please alway provide meaningful defaults for everything :)